nav2_smac_planner/src/smac_planner_2d.cpp

```diff
@@ -340,45 +334,46 @@ nav_msgs::msg::Path SmacPlanner2D::createPlan(
   return plan;
 }
 
-void SmacPlanner2D::on_parameter_event_callback(
-  const rcl_interfaces::msg::ParameterEvent::SharedPtr event)
+rcl_interfaces::msg::SetParametersResult
+SmacPlanner2D::dynamicParametersCallback(std::vector<rclcpp::Parameter> parameters)
 {
+  auto result = rcl_interfaces::msg::SetParametersResult();
```

Comment:
err, umm maybe `rcl_interfaces::msg::SetParametersResult result` instead? I think you've been writing some python code recently :wink:

Reply:
hum, at least from cpp 11 using `auto` is pretty convenient, and I was just reproducing rclcpp style:
https://github.com/ros2/rclcpp/blob/d5ec2580805c3abd8b6fecaf3d09040e00e9f15e/rclcpp/src/rclcpp/parameter_service.cpp#L86
But no problem to use `rcl_interfaces::msg::SetParametersResult result` instead if you want

Reply:
Do `rcl_interfaces::msg::SetParametersResult result;`, its more compact. Clearly whoever wrote that was thinking in python. Its the same thing without `auto` or `()`, making it more compact. There's no reason to use auto for the default constructor.

Reply:
:man_facepalming: nice catch, did not realize it. Will change at next push